core/recipe-compiler/src/test/kotlin/RecipeCompilerTest.kt

```diff
@@ -68,4 +68,21 @@ class RecipeCompilerTest {
         assertEquals("796a3cb3eb68b35d", compiled.recipeId())
     }
 
+
+    @Test
+    fun `should compile dsl to recipe ingredient`() {
+
+        val recipe = recipe("recipe") {
+            sensoryEvents {
+                event<EventA>()
+            }
+            interaction<InteractionA>()
+            ingredient<String, String>("extract"){"Hello123"}
+        }
```

Comment:
Maybe consider renaming this `ingredient` to `ingredientTransformation` of `ingredientMapping`, since that might explain its purpose a bit better.